docs(contributing): Fix the ROADMAP.md uri

tim-mcgee opened the pull request:
<!--
Thanks for your interest in the project. I appreciate bugs filed and PRs submitted!

Please make sure that you are familiar with and follow the Code of Conduct for this project (found in the CODE_OF_CONDUCT.md file).

Also, please make sure you're familiar with and follow the instructions in the contributing guidelines (found in the CONTRIBUTING.md file).

If you're new to contributing to open source projects, you might find this free video course helpful: http://kcd.im/pull-request

Please fill out the information below to expedite the review and (hopefully) merge of your pull request!
-->

<!-- What changes are being made? (What feature/bug is being fixed here?) -->
**What**: I fixed the link to the ROADMAP.md file in the CONTRIBUTING.md file

<!-- Why are these changes necessary? -->
**Why**: Because the link wasn't working correctly

<!-- How were these changes implemented? -->
**How**: With a Text-Editor?


<!-- feel free to add additional comments -->
I started at a new feature similar to avas or tapes --require to specify a module which will be required before the actual config so the config could be written in es7 while I noticed the link was broken.

kentcdodds commented:
Sweet. One thing.

kentcdodds commented on .all-contributorsrc:
```diff
@@ -57,6 +57,17 @@
+      "contributions": [
+        "code",
+        "doc",
+        "test"
```
Looks like this contribution is only `doc`, could you remove `code` and `test`? (Or contribute code/tests) :-)

tim-mcgee replied:
Just added code as well :)

tim-mcgee commented on bin/p-s.js:
```diff
@@ -46,7 +47,7 @@ runPackageScript({
 }, function(result) {
   if (result.error) {
     log.error(result.error)
-    const FAIL_CODE = 1
```
This file isn't compiled so it has to be Node 0.10.0 friendly. Const is only available behind a flag in Node 0.10.0 [node.green](http://node.green/#const)

kentcdodds replied:
Yikes! Thanks for noticing that! Would you like to [makeapullrequest.com](http://makeapullrequest.com)?

kentcdodds replied:
:+1:

tim-mcgee commented on bin/p-s.js:
```diff
@@ -55,7 +56,22 @@ runPackageScript({
+      }
+    }
+    var config = require(psConfigFilename)
+    if (programm.resolve === 'babel-register') {
```
The problem with babel-register is that you get the default export behind a module with a default property. This line fixes it but I think it's not really pretty :)

kentcdodds commented:
This LGTM. Could anyone else weigh in?

kentcdodds commented on bin/p-s.js:
```diff
@@ -55,7 +56,23 @@ runPackageScript({
+    }
+
+    var config = require(psConfigFilename)
+    if (program.require === 'babel-register') {
```
Could we actually do:

```javascript
if (config.default) {
  config = config.default
}
```

That should be safer and allow people to use ES6 modules or CommonJS even if they do transpile with babel or anything else.

kentcdodds replied:
Would probably be good to add a comment there as well.

kentcdodds commented:
Oh, and also... About tests.... Yeah, I'm not sure how to test this... We can leave it off for now.

kentcdodds commented on bin/p-s.js:
```diff
@@ -55,7 +56,23 @@ runPackageScript({
+      } catch (e) {
+        log.warn({
+          message: colors.yellow('Unable to preload ' + program.require),
+          ref: 'preload-module'
```
This `ref` should reference a line in [ERRORS_AND_WARNINGS.md](https://github.com/kentcdodds/p-s/blob/master/other/ERRORS_AND_WARNINGS.md). Similar to how [this](https://github.com/kentcdodds/p-s/blob/9ae2217d365c06153d47db2acfd63230fcff83fb/bin/p-s.js#L62) references [this](https://github.com/kentcdodds/p-s/blob/9ae2217d365c06153d47db2acfd63230fcff83fb/other/ERRORS_AND_WARNINGS.md#unable-to-find-config). If you could add a line to ERRORS_AND_WARNINGS.md, that'd be great 👍

kentcdodds commented:
I merged your other PR. You'll need to [rebase this branch](http://kcd.im/pr-rebase).

tim-mcgee commented:
I hope I did all things right :D

tim-mcgee commented on README.md:
```diff
@@ -290,8 +295,9 @@ benefits of npm scripts I wanted to keep).
-| :---: | :---: | :---: | :---: | :---: | :---: |
+| [<img src="https://avatars.githubusercontent.com/u/1500684?v=3" width="100px;"/><br /><sub>Kent C. Dodds</sub>](http://kent.doddsfamily.us)<br />[💻](https://github.com/kentcdodds/p-s/commits?author=kentcdodds) [📖](https://github.com/kentcdodds/p-s/commits?author=kentcdodds) 🚇 💡 | [<img src="https://avatars.githubusercontent.com/u/532272?v=3" width="100px;"/><br /><sub>David Wells</sub>](http:// ...
+
+-| :---: | :---: | :---: | :---: | :---: | :---: |
```
I failed while rebasing... I hope the list is fixed now.

kentcdodds replied:
If you preview this, it looks like it's formatted incorrectly: https://github.com/tim-mcgee/p-s/blob/98aaff50503ec1811a6da755ab6fd7f0b085f361/README.md

Could you run `npm start generateContributors` to automatically update the table?

kentcdodds commented on other/ERRORS_AND_WARNINGS.md:
```diff
@@ -18,3 +18,11 @@ point to a file that actually exists.
+
+## Unable to preload module
+
+This happens the module which you try to preload couldn't be resolved.
```
This could probably more clearly call out the relationship with the `--require` flag. Something like:

> This happens when you use the `--require` flag and the module you specify cannot be resolved.

kentcdodds commented on other/ERRORS_AND_WARNINGS.md:
```diff
@@ -18,3 +18,11 @@ point to a file that actually exists.
+
+### To fix:
+
+Check if you spelled the module correctly.
```
I would say:

> 1. Check that you spelled the module correctly
> 2. Check that the module you wish to require is require-able

kentcdodds commented on bin/p-s.js:
```diff
@@ -55,7 +56,25 @@ runPackageScript({
+    // Require a module like babel-register
+    if (program.require) {
+      try {
+        require(program.require)
```
If someone wanted to `--require ./some-file/in/their-project.js` this wouldn't work. I think there's a way to use `resolve` to make this work relative to `process.cwd()`.

tim-mcgee replied:
if (program.require.substr(0, 0) === '.') {
  program.require = path.resolve(process.cwd(), program.require)
}

kentcdodds replied:
No, because they could do an absolute path or a relative path without a dot too

tim-mcgee replied:
But if you require('some-file/in/your-project.js') node would cry too? So I think the only problem would be resolving the relative paths because I you don't need to resolve absolute paths because their already absolute and path.resolve('/some/absolute/path.js') would return '/some/absolute/path.js' anyway. Or I am wrong at this?

The other option I could imagine is Module.prototype.require.call(process.cwd(), programm.require) how about this?

kentcdodds replied:
Good point. Yeah, let's just check if it starts with `.` 👍

kentcdodds commented:
I added you as a computer. Please feel free to continue to file PRs as you see fit. 😃

kentcdodds commented:
Also, looks like this branch has fallen behind again and needs to be rebased. Sorry!

tim-mcgee commented:
I added the `'.'` check and also added the options to the `src/bin-utils.js` file.
I tested absolute, relative and npm modules manually. Should work fine :)

kentcdodds commented:
This looks perfect! Thanks @tim-mcgee!

kentcdodds commented:
I'd like to get at least one other person to review. @rowanoulton, @abhishekisnot, or @giladgo, could one of you look at this and merge it?

tim-mcgee commented on bin/p-s.js:
```diff
@@ -55,7 +56,29 @@ runPackageScript({
+    var config = require(psConfigFilename)
+    // babel-register is loading the default package into a object with the prop
+    // default so if it exist use it as the config
+    if (config.default) {
```
Maybe we should check `if (config.__esModule === true)` instead of `if (config.default)`?

kentcdodds replied:
Ah, yeah, that's what Babel does after all:

```js
'use strict';

var _x = require('x');

var _x2 = _interopRequireDefault(_x);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
```

tim-mcgee commented:
Maybe we should sometimes exchange all the underlying code so it works like babel so you could do named export as well?

````javascript
const scripts = {...}
const options = {...}

export {scripts, options} // hope this is the right syntax I am mainly using default exports...
````

kentcdodds commented:
> Maybe we should sometimes change all the underlying code so it works like babel and you could do named exports as well?

Nah, let's keep it simple. Let's try to avoid multiple ways of doing the same thing.

giladgo commented on bin/p-s.js:
```diff
@@ -55,7 +56,29 @@ runPackageScript({
-    return require(psConfigFilename)
+    // Require a module like babel-register
+    if (program.require) {
+      var requirePath = (program.require.substr(0, 1) === '.'
```
isn't `program.require[0]` simpler?

giladgo commented:
Other than one small, comment, LGTM

kentcdodds commented:
Awesome! Thanks for working on this! :shipit:
